### DISCUSSION TITLE
Wind speed / estimator - does it work ? (OSD shows "---"  on my F722-STD quad)

### DISCUSSION THREAD

**Discussion by alccrl** [iNavFlight/inav#6718] [Q&A] [answered]
Hi,
Wind speed is shown on OSD, after offline configuration addition to OSD, and legenda is shown through 5.8 Ghz goggles, but there is no numeric value for horizonatl/vertical speed, only "---" is shown.
I use Inav 2.6.1 and Mateksys F722-STD (RET6) , baro, compass, gps, rth, failsafe, hold, everything (else) works.
Looking at the source code of inav-2.6.1 I cannot at the moment find a rational  (I am new to inav sw, so no experience on it, for now).
Does anybody have some indication about this ?

Alfredo
![Wind speed not shown](https://user-images.githubusercontent.com/78967144/111458186-71510780-8719-11eb-872d-c255f96d9ad8.jpg)

**Comment by avsaase** [comment, marked as answer]
Wind estimation is currently not possible on multirotors, only airplanes.

**Reply by alccrl** [reply]
Thanks for Your immediate answer !. 
Do You know where fixed_wing or quad are discriminated in the source code ? It would me help in understanding the flows.
 Alfredo

**Reply by alccrl** [reply]
I read an old post about this, but it did not convince me very much, I think that a rought indication of wind should be possible, at least during rth, being clear the direction and the parameters (setup in configurator) the quad goes to.
This is not a basic feature, of course, but flying among the mountains as I do, it would help decision on flight route to take.

**Reply by avsaase** [reply]
> Thanks for Your immediate answer !.
> Do You know where fixed_wing or quad are discriminated in the source code ? It would me help in understanding the flows.
> Alfredo

I suspect it is here:
https://github.com/iNavFlight/inav/blob/3c97e916fed96e87cb3410f60b9f6214287340af/src/main/flight/wind_estimator.c#L82-L87

I don't fully understand the wind estimator, but I think it relies on the fact that airplanes can only move forward, so it uses the difference between the fuselage direction changes and the GPS direction changes to estimate the wind speed. MR don't have this property so it would need a different algorithm. 

It should be noted that the wind estimation is not very accurate. I think ArduPilot has pretty accurate wind speed estimation and uses that for virtual airspeed. It would be nice to port that to INAV at some point. I have looked into it and I think the wind estimator is part of a larger system that is much more complex than INAV, so it may not be possible to port just the wind estimator.

**Reply by alccrl** [reply]
I found that code too, using eclipse-cdt for browsing,  and I compiled (I was curious) the 2.7.0 source form github using build.sh script (under Linux).
Unfornutately I did not succed in compiling using eclipse-cdt, I am not expert of docker boxes, so I configured some #define and #includes by hand looking at the generated Makefiles for MATEKF722. This helps me navigating the code.
While navigating wind estimator, I found some interesting code about RTH that will help me understand better what happened to my quad that recently exausted the battery while landing durin RTH, reset the controller, but unexpectely rearmed spontaneosly and slowed down the fall, avoid a complete disaster (I only destroyed the vtx antenna).
I yet have to look at the black box logs, but the audio of my runcam-4k is quite self-explanatory about what happened.
I think that wind estimator and rth may be related someway each other, during quad rth at least, not only good for fixed wing.

Alfredo
